src/MSBuild/XMake.cs

```diff
@@ -3486,7 +3679,7 @@ private static void ProcessBinaryLogger(string[] binaryLoggerParameters, List<IL
             // If we have a binary logger, force verbosity to diagnostic.
             // The only place where verbosity is used downstream is to determine whether to log task inputs.
             // Since we always want task inputs for a binary logger, set it to diagnostic.
-            verbosity = LoggerVerbosity.Diagnostic;
+            verbosity = canAdjustVerbosity ? LoggerVerbosity.Diagnostic : verbosity;
```

Comment:
How is this exactly used?
I can imagine someone might want to evaluate properties (and see them in output), while at the same time troubleshoot the processing with the binary log (or file log) - is that possible?
Or even - I can imagine future feature request for outputing those into a user speicifed file - so console logger then might be just fine

Reply:
Hmm...@baronfel, I don't think someone would want to see property values while also gathering a binlog, since this would be getting something like OutputPath before running a second build and using the result here in that build. Would people actually want a binlog from this "get property" build?

Reply:
IMO no, if necessary in an emergency we can always fall back on `MSBUILDDEBUGENGINE`.

Reply:
The reason I did this is that this change is intended to suppress _all_ output if a user sets one of the flags...except the output from this change. This part prevents unintended output if, say, the user uses a script to build their repo that includes /bl but also adds one of these parameters to the command line. In that case, I would think we'd want to suppress the output anyway, so we wouldn't want the /bl to be properly respected if that makes sense.

Reply:
I think it's fair to see this eval feature as exclusive with binlogs - if someone has/uses a binlog then they can code the moral equivalent of this feature, and this feature is very strongly focused on the CLI invocation use case only.